### ISSUE TITLE
Consider adding a toPosix function

### ISSUE THREAD

**Issue by kutyel** [justinmimbs/elm-date-basic#27]
Hi!

Thanks for the `Date` type, it definitely makes my life easier!

However, when dealing with Firestore stuff with elm I had the need to convert dates to `Posix`, would you consider adding this function to the library?

```elm
toPosix : Date -> Posix
```

Thanks and keep up the good work! 💪🏻

**Comment by malaire** [comment]
Since `Date` represents a date in specific timezone (given as parameter to `fromPosix`), conversion back will also require timezone and can result in 0, 1 or 2 `Posix` values.

So `toPosix : Date -> Posix` is not possible and it would need to be something like

    toPosix : Zone -> Date -> List Posix

Alternatively you could just add `utcToPosix : Date -> Posix` which assumes that `Date` is in UTC.

**Comment by turboMaCk** [comment]
I agree that lack of this is kind of problematic. While dealing with apis which uses unix timestamps there is a nice way to go from the time stamp to `Date` there is no nice way to go from `Date` back to timestamp which is often needed.

The simplest workaround at the moment is to go through the iso8601 and deal with Result.

```elm
Iso8601.toTime << Date.toIsoString
```

**Comment by justinmimbs** [comment]
Hi there,

You can convert a `Date` to `Posix` using [`Time.Extra.partsToPosix`](https://package.elm-lang.org/packages/justinmimbs/time-extra/latest/Time-Extra#partsToPosix), as I mentioned in [this issue](https://github.com/justinmimbs/date/issues/24).

Here's an example of such a function.

```elm
dateToPosix : Time.Zone -> Date -> Posix
dateToPosix zone date =
    Time.partsToPosix zone
        { year = Date.year date
        , month = Date.month date
        , day = Date.day date
        , hour = 0
        , minute = 0
        , second = 0
        , millisecond = 0
        }
```

I hope this helps.

Justin

**Comment by kutyel** [comment]
Closed as a duplicate of #24, thanks @justinmimbs!
